libraries/script-engine/src/ScriptEngine.cpp

```diff
@@ -645,37 +644,38 @@ void ScriptEngine::run() {
+                    audioPacket->writePrimitive(SCRIPT_AUDIO_BUFFER_SAMPLES);
 
                     // use the orientation and position of this avatar for the source of this audio
-                    packetStream.writeRawData(reinterpret_cast<const char*>(&_avatarData->getPosition()), sizeof(glm::vec3));
+                    audioPacket->writePrimitive(_avatarData->getPosition());
                     glm::quat headOrientation = _avatarData->getHeadOrientation();
-                    packetStream.writeRawData(reinterpret_cast<const char*>(&headOrientation), sizeof(glm::quat));
+                    audioPacket->writePrimitive(headOrientation);
 
                 } else if (nextSoundOutput) {
                     // assume scripted avatar audio is mono and set channel flag to zero
-                    packetStream << (quint8)0;
+                    audioPacket->writePrimitive((quint8) 0);
 
                     // use the orientation and position of this avatar for the source of this audio
-                    packetStream.writeRawData(reinterpret_cast<const char*>(&_avatarData->getPosition()), sizeof(glm::vec3));
+                    audioPacket->writePrimitive(_avatarData->getPosition());
                     glm::quat headOrientation = _avatarData->getHeadOrientation();
-                    packetStream.writeRawData(reinterpret_cast<const char*>(&headOrientation), sizeof(glm::quat));
+                    audioPacket->writePrimitive(headOrientation);
 
                     // write the raw audio data
-                    packetStream.writeRawData(reinterpret_cast<const char*>(nextSoundOutput), numAvailableSamples * sizeof(int16_t));
+                    audioPacket->write(reinterpret_cast<const char*>(nextSoundOutput), numAvailableSamples * sizeof(int16_t));
                 }
-                
+
                 // write audio packet to AudioMixer nodes
                 auto nodeList = DependencyManager::get<NodeList>();
-                nodeList->eachNode([this, &nodeList, &audioPacket, &numPreSequenceNumberBytes](const SharedNodePointer& node){
+                nodeList->eachNode([this, &nodeList, &audioPacket](const SharedNodePointer& node){
                     // only send to nodes of type AudioMixer
                     if (node->getType() == NodeType::AudioMixer) {
                         // pack sequence number
                         quint16 sequence = _outgoingScriptAudioSequenceNumbers[node->getUUID()]++;
-                        memcpy(audioPacket.data() + numPreSequenceNumberBytes, &sequence, sizeof(quint16));
-                        
+                        audioPacket->seek(0);
+                        audioPacket->writePrimitive(sequence);
+
                         // send audio packet
-                        nodeList->writeDatagram(audioPacket, node);
+                        nodeList->sendPacket(std::move(audioPacket), node);
```

Comment:
It looks like we are passing ownership of the NLPacket here for each AudioMixer - I think we need to make a copy or use sendUnreliablePacket here